### PR TITLE
allow sending raw commands to devices

### PR DESCRIPTION
**Pull request by lnr0626**
## Proposed change
This arose while debugging #411 (and messing around with the other commands I found in the command tables). Essentially I found it a lot easier to debug the actual commands when I had the ability to send raw insteon commands through mqtt

I don't think this is a common use case, and there's some cleanup that needs to happen if this were desired (tests, docs, etc); but am happy to work through those if this is would be desired

## Checklist
- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] Tests have been added to verify that the new code works.
- [x] Code documentation was added where necessary

If user exposed functionality or configuration variables are added/changed:

- [x] Documentation added/updated

### CODE REVIEW

**Comment by krkeegan** [comment]
I can understand how something like this would be helpful.

I think you should allow a full extended message to be defined, not just all zero bytes.  That gets a bit unweildy from the command line, but from a json mqtt message, the extended bytes can just be a list.

**Comment by lnr0626** [comment]
It does allow full extended commands - the mqtt message takes a data list, and the cli takes a `-d`/`--data` argument that's zero padded and trimmed to 14 bytes - the `--ext-req` flag is purely a convenience shorthand :-)

**Comment by krkeegan** [comment]
Ahh, I messed that. That is perfect then.

Yes, this would have been quite helpful at times.  I had failed to realize it could be added so easily.

File: insteon_mqtt/device/base/Base.py [hunk 1]
```diff
@@ -493,6 +494,31 @@ def set_flags(self, on_done, **kwargs):
...
+        This can be used to send commands that aren't supported directly
+        by insteon-mqtt yet. This is as low level as it gets, use at your
+        own risk
+        """
+        if data is None:
+            msg = Msg.OutStandard.direct(self.addr, cmd1, cmd2)
+        else:
+            msg = Msg.OutExtended.direct(self.addr, cmd1, cmd2, bytes(data), crc_type)
+
+        if ext_resp:
```
Comment (lnr0626) [inline insteon_mqtt/device/base/Base.py:512]: Haha whoops, this condition is backwards

File: insteon_mqtt/device/base/Base.py [hunk 2]
```diff
@@ -1149,6 +1175,16 @@ def handle_flags(self, msg, on_done):
                "{:08b}".format(msg.cmd2))
         on_done(True, "Operation complete", msg.cmd2)
 
+    #-----------------------------------------------------------------------
+    def handle_std_raw_message(self, msg, on_done):
```
Comment (lnr0626) [inline insteon_mqtt/device/base/Base.py:1179]: I'll probably just combine these handlers, I want sure if the response/signature was different based on standard/extended messages

**Comment by lnr0626** [comment]
Cool beans, I'll add some docs and tests to get this ready to merge.

I'll also move the pad/trim logic to the message handler, and may switch the cmd name to `raw_command` or `raw` - do you have a preference?

**Comment by krkeegan** [comment]
I don't have a name preference.  I can see a value in either.

**Comment by krkeegan** [comment]
Looks good to me, unless you have any more changes I will merge.

**Comment by lnr0626** [comment]
I think it's good to go, feel free to merge away :-)

**Comment by krkeegan** [comment]
This works in my setup. I didn't test extensively, but a basic test of standard commands worked as expected.  Thank you for this, it is very helpful.

**Pull request merged**